src/openforms/prefill/api/views.py

```diff
@@ -67,3 +76,73 @@ def get_objects(self):
         choices = plugin.get_available_attributes()
 
         return [ChoiceWrapper(choice) for choice in choices]
+
+
+@extend_schema(
+    summary=_("List available objecttypes for Objects API"),
+    parameters=[OBJECTS_API_GROUP_QUERY_PARAMETER],
+)
+class PluginObjectsAPIObjecttypeListView(ObjecttypesListView):
+    """
+    List the available prefill objecttypes for Objects API plugin.
+    """
+
+    pass
+
+
+@extend_schema(
+    summary=_("List available objecttype versions for Objects API"),
+    parameters=[OBJECTS_API_GROUP_QUERY_PARAMETER],
+)
+class PluginObjectsAPIObjecttypeVersionListView(ObjecttypeVersionsListView):
+    """
+    List the available prefill objecttype versions for Objects API plugin.
+    """
+
+    def get_objects(self):
+        input_serializer = ObjectsAPIGroupInputSerializer(
+            data=self.request.query_params
+        )
+        input_serializer.is_valid(raise_exception=True)
+
+        config_group = input_serializer.validated_data["objects_api_group"]
+        objecttype_uuid = self.kwargs["objects_api_objecttype_uuid"]
+
+        with get_objecttypes_client(config_group) as client:
+            return client.list_objecttype_versions(objecttype_uuid)
+
+
+@extend_schema(
+    summary=_("List available attributes for Objects API"),
+    parameters=[OBJECTS_API_GROUP_QUERY_PARAMETER],
+)
+class PluginObjectsAPIAttributesListView(ListMixin, APIView):
+    """
+    List the available attributes for Objects API plugin.
+    """
+
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+    serializer_class = PrefillAttributeSerializer
+
+    def get_objects(self):
+        plugin = register["objects_api"]
+        input_serializer = ObjectsAPIGroupInputSerializer(
+            data=self.request.query_params
+        )
+        input_serializer.is_valid(raise_exception=True)
+
+        config_group = input_serializer.validated_data["objects_api_group"]
+        choices = plugin.get_available_attributes(
+            reference={
+                "objects_api_group": config_group,
+                "objects_api_objecttype_uuid": self.kwargs[
+                    "objects_api_objecttype_uuid"
+                ],
+                "objects_api_objecttype_version": self.kwargs[
+                    "objects_api_objecttype_version"
+                ],
+            }
+        )
```

Comment:
I don't think this needs to be a method on the `Plugin` class.

You want something similar here like `TargetPathsListView` in the registration plugin - all the possible properties from the Object Type JSON schema definition should be returned here so you can select it in a dropdown for the mappings. It's important that the schema definition (or part of it at least) is also returned, so that on the client side you can validate/match up things (e.g. a property in the objecttype that is `type: object` cannot be assigned to a textfield form variable which has `type: string`).

I'm really expecting a list of objects that serialize with https://github.com/open-formulieren/open-forms/blob/beacb481e2dc31ec5fc9cfe56c157420b1f42d80/src/openforms/registrations/contrib/objects_api/api/serializers.py#L40 (`target_path` and `json_schema` are the relevant fields here)

You will need to do JSON schema processing here in the view, but I think you can just re-use https://github.com/open-formulieren/open-forms/blob/beacb481e2dc31ec5fc9cfe56c157420b1f42d80/src/openforms/registrations/contrib/objects_api/api/views.py#L119

The big difference is that we don't have a schema on our end yet to filter down the possible options, so those target paths input serializer usages are not relevant for prefill.